[SPARK-36154][DOCS] Documenting week and quarter as valid formats in pyspark sql/functions trunc

dominikgehl opened the pull request:
### What changes were proposed in this pull request?
Added missing documentation of week and quarter as valid formats to pyspark sql/functions trunc

### Why are the changes needed?
Pyspark documentation and scala documentation didn't mentioned the same supported formats

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Only documentation change

dominikgehl committed:
documenting week and quarter as valid formats

AmplabJenkins commented:
Can one of the admins verify this patch?

HyukjinKwon commented on python/pyspark/sql/functions.py:
```diff
@@ -2042,7 +2042,7 @@ def trunc(date, format):
     date : :class:`~pyspark.sql.Column` or str
     format : str
-        'year', 'yyyy', 'yy' or 'month', 'mon', 'mm'
+        'year', 'yyyy', 'yy' or 'month', 'mon', 'mm', or 'week' or 'quarter'
```
```suggestion
        'year', 'yyyy', 'yy', 'month', 'mon', 'mm', 'week' or 'quarter'
```

HyukjinKwon replied:
Actually, can we copy the docs in from Scala's trunc? 

also should fix https://github.com/apache/spark/blob/b5ee6d008d9cc3ad5061ad7415f7099e27769853/R/pkg/R/functions.R#L49-L51 too

dominikgehl replied:
I took the Scala doc

dominikgehl committed:
documentation change

dominikgehl committed:
using scala doc

HyukjinKwon commented:
can you update https://github.com/apache/spark/blob/b5ee6d008d9cc3ad5061ad7415f7099e27769853/R/pkg/R/functions.R#L49-L51 too while we're here

dominikgehl committed:
R documentation and date_trunc

dominikgehl commented:
> can you update
> 
> https://github.com/apache/spark/blob/b5ee6d008d9cc3ad5061ad7415f7099e27769853/R/pkg/R/functions.R#L49-L51
> 
> too while we're here

Sure. Updated and also fixed date_trunc at the same time

HyukjinKwon commented:
ok to test

SparkQA commented:
**[Test build #141078 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/141078/testReport)** for PR 33359 at commit [`be07fba`](https://github.com/apache/spark/commit/be07fbaea5f76fb7b4cd9851a21622f6d521205b).

SparkQA commented:
Kubernetes integration test starting
URL: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/45592/

SparkQA commented:
**[Test build #141078 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/141078/testReport)** for PR 33359 at commit [`be07fba`](https://github.com/apache/spark/commit/be07fbaea5f76fb7b4cd9851a21622f6d521205b).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/141078/

SparkQA commented:
Kubernetes integration test status success
URL: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/45592/

AmplabJenkins commented:
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/45592/

MaxGekk commented:
+1, LGTM. Merging to master/3.2/3.1/3.0.
Thank you, @dominikgehl and @HyukjinKwon @nishaantt for review.
